src/helics/core/CoreBroker.cpp

```diff
@@ -2703,6 +2716,43 @@ std::string CoreBroker::generateQueryAnswer(const std::string& request, bool for
     return "#invalid";
 }
 
+std::string CoreBroker::generateGlobalStatus(JsonMapBuilder& builder)
+{
+    auto cstate = generateQueryAnswer("current_state", false);
+    auto jv = loadJsonStr(cstate);
+    std::string state;
+    if (jv["federates"][0].isObject()) {
+        state = jv["state"].asString();
+    } else {
+        state = "init_requested";
+    }
+    if (state == "error") {
+        Json::Value v;
+        v["status"] = "error";
+        v["timestep"] = -1;
+        return generateJsonString(v);
+    }
+
+    Time mv{Time::maxVal()};
+    if (!builder.getJValue()["cores"][0].isObject()) {
+        state = "init_requested";
+    }
```

Comment:
I'm not seeing where `state` is getting used after it gets used here -- is this needed or should there be a condition wrapping the subsequent for loop that checks state?